core/src/mindustry/input/InputHandler.java

```diff
@@ -601,6 +617,44 @@ protected void removeSelection(int x1, int y1, int x2, int y2, boolean flush){
         }
     }
 
+    protected void updateArea(int x1, int y1, int x2, int y2, boolean forced){
+        lineRequests.clear();
+
+        NormalizeResult dresult = Placement.normalizeArea(x1, y1, x2, y2, rotation, false, maxLength);
+        for(int x = dresult.x; x <= dresult.x2; x++){
+            for(int y = dresult.y; y <= dresult.y2; y++){
+                Tile tile = world.tilec(x, y);
+                if(tile.block() == Blocks.air || tile.block() instanceof Rock
+                    || !validPlace(x, y, block, tile.rotation())
+                    || !(tile.block() instanceof Conveyor || tile.block() instanceof Conduit || tile.block() instanceof Wall)) continue;
+
```

Comment:
ima just leave this comment from anuke here: https://github.com/Anuken/Mindustry/pull/950#discussion_r341330988

Reply:
I'll try to use it in this implementation. Thank you for guidance.